py/client/pydeephaven/session.py

```diff
@@ -181,20 +249,26 @@ def close(self) -> None:
     def release(self, ticket):
         self.session_service.release(ticket)
 
-    def _parse_script_response(self, response):
-        if response.changes.created:
-            for t in response.changes.created:
-                t_type = None if t.typed_ticket.type == '' else t.typed_ticket.type
-                self._tables[t.field_name] = (t_type, Table(session=self, ticket=t.typed_ticket.ticket))
+    def _parse_fields_change(self, fields_change, allow_scope_changes):
+        if fields_change.created:
+            for t in fields_change.created:
+                if allow_scope_changes or t.application_id != 'scope':
+                    t_type = None if t.typed_ticket.type == '' else t.typed_ticket.type
+                    self._fields[t.field_name] = (t_type, Table(session=self, ticket=t.typed_ticket.ticket))
+
+        if fields_change.updated:
+            for t in fields_change.updated:
+                if allow_scope_changes or t.application_id != 'scope':
+                    t_type = None if t.typed_ticket.type == '' else t.typed_ticket.type
+                    self._fields[t.field_name] = (t_type, Table(session=self, ticket=t.typed_ticket.ticket))
 
-        if response.changes.updated:
-            for t in response.changes.updated:
-                t_type = None if t.typed_ticket.type == '' else t.typed_ticket.type
-                self._tables[t.field_name] = (t_type, Table(session=self, ticket=t.typed_ticket.ticket))
+        if fields_change.removed:
+            for t in fields_change.removed:
+                if allow_scope_changes or t.application_id != 'scope':
+                    self._fields.pop(t.field_name, None)
 
-        if response.changes.removed:
-            for t in response.changes.removed:
-                self._tables.pop(t.field_name, None)
+    def _parse_script_response(self, response):
+        self._parse_fields_change(response.changes, True)
```

Comment:
I like this idea, but I think it should be flipped around. To me it seems most intuitive that the user's own operations should always be immediately accessible (and the most convenient); being able to get other sessions' tables should be an opt-in (this whole feature is an opt-in itself, after all). I think the "I need everyone's tables, but asynchronously" usecase is fairly niche.

```py
session1.sync_fields(True)
async_changes = session1.async_changes_handle()

session1.run_script('t = empty_table(1)')
x1 = session1.open_table('t') # works immediately
x2 = async_changes_handle.open_table('t') # eventual access if sync'd
```

Though, while we're throwing out ideas: What about a separate `AsyncSession` class with the same interface but different guarantees? No idea how it would work, but just throwing it out.